File: 240410/블럭쌓는 명령2/block-stacking-commands2.py
N, K = map(int, input().split())
blocks = []
for _ in range(K):
    first, second = map(int, input().split())
    blocks.append([first, second])


checked = [0] * (N + 1)
for start, end in blocks:
    for i in range(start, end + 1):
        checked[i] += 1

print(max(checked))

File: 260118/블럭쌓는 명령2/block-stacking-commands2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, K;
    cin >> N >> K;

    vector<pair<int,int>> blocks(K);
    for (int i = 0; i < K; i++) 
    {
        int first, second;
        cin >> first >> second;
        blocks[i] = {first, second};
    }

    // Please write your code here.
    vector<int> checked(N + 1, 0);
    for (auto [start, end] : blocks) 
    {
        for (int i = start; i <= end; i++) 
        {
            checked[i] += 1;
        }
    }

    cout << *max_element(checked.begin(), checked.end()) << "\n";
    return 0;
}